core/domain/activity_jobs_one_off.py

```diff
@@ -180,3 +182,48 @@ def map(commit_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class FixCommitLastUpdatedOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that sets the last_updated in *CommitLogEntryModels to created_on if
+    the last_updated is in the timespan when user ID migration was done.
+    """
+
+    MIGRATION_START = datetime.datetime.strptime(
+        '2020-06-29T07:00:00Z', '%Y-%m-%dT%H:%M:%SZ')
+    MIGRATION_END = datetime.datetime.strptime(
+        '2020-06-29T13:00:00Z', '%Y-%m-%dT%H:%M:%SZ')
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(FixCommitLastUpdatedOneOffJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [
+            collection_models.CollectionCommitLogEntryModel,
+            exp_models.ExplorationCommitLogEntryModel,
+            question_models.QuestionCommitLogEntryModel,
+            skill_models.SkillCommitLogEntryModel,
+            story_models.StoryCommitLogEntryModel,
+            topic_models.TopicCommitLogEntryModel,
+            topic_models.SubtopicPageCommitLogEntryModel
+        ]
+
+    @staticmethod
+    def map(commit_model):
+        class_name = commit_model.__class__.__name__
+        last_updated = commit_model.last_updated
+        if (FixCommitLastUpdatedOneOffJob.MIGRATION_START < last_updated <
+                FixCommitLastUpdatedOneOffJob.MIGRATION_END):
+            commit_model.last_updated = commit_model.created_on
+            commit_model.put(update_last_updated_time=False)
+            yield ('SUCCESS_FIXED - %s' % class_name, commit_model.id)
+        else:
+            yield ('SUCCESS_ALREADY_CORRECT - %s' % class_name, commit_model.id)
```

Comment:
Can you maybe divide this into two parts: (a) the created on time is close to the last updated time (in that case, just need the count of values), and (b) the created on time is at least a week different from the last updated time (in that case, provide details like the id and model type so we can do a spot-check).

Reply:
Done. The incorrect (they can be correct, just a naming used in the MapReduce output) commits are these that have `last_updated` more than a week later than `created_on` or these that have `last_updated` earlier than `created_on`.